components/UserAuth/ResetPassword.tsx

```diff
@@ -35,46 +42,48 @@ export default function ResetPassword({ token }) {
     },
   )
 
+  const successfulError = data?.redeemUserPasswordResetToken?.code
+    ? data?.redeemUserPasswordResetToken
+    : undefined
+
   const handleSubmit = async (event) => {
     event.preventDefault()
     await passwordReset()
     resetForm()
   }
 
-  //@TODO: add error handling
   return (
-    <>
-      <h2>Reset your password</h2>
-      <form method="POST" onSubmit={handleSubmit}>
-        <fieldset disabled={loading}>
-          {data?.redeemUserPasswordResetToken === null && (
-            <p>Success! You can Now sign in</p>
-          )}
-          <label htmlFor="email">
-            Email
-            <input
-              type="email"
-              name="email"
-              placeholder="Email address"
-              autoComplete="email"
-              value={inputs.email}
-              onChange={handleChange}
-            />
-          </label>
-          <label htmlFor="password">
-            New password
-            <input
-              type="password"
-              name="password"
-              placeholder="********"
-              autoComplete="password"
-              value={inputs.password}
-              onChange={handleChange}
-            />
-          </label>
-          <button type="submit">Reset Password</button>
-        </fieldset>
-      </form>
-    </>
+    <FormStyle method="POST" onSubmit={handleSubmit}>
+      <FormHeader>Reset your password</FormHeader>
+      <DisplayError error={error || successfulError} />
+      <FieldSetStyle disabled={loading}>
+        {data?.redeemUserPasswordResetToken === null && (
```

Comment:
`null` is a false value... or are you checking specific for `null`? That means `false` or `undefined` is not catched here